### ISSUE TITLE
failed to install unar

### ISSUE THREAD

**Issue by axzxc1236** [Homebrew/linuxbrew-core#14705]
**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [ ] are reporting a bug others will be able to reproduce and not asking a question or requesting software. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh. To get software added or changed in Homebrew please file a [Pull Request](https://github.com/Homebrew/linuxbrew-core/blob/master/CONTRIBUTING.md)

**I don't really sure if someone else is able to reproduce it, but this happened on a almost fresh VPS, installed Ubuntu 18.04.1 and then updated to 18.04.3**

- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, Homebrew/homebrew-core formula (not cask) on macOS? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a Linux problem please file this issue at https://github.com/Homebrew/linuxbrew-core/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?

https://gist.github.com/6bfab7cbdf79d9091f9904e3b62f6b17

- [ ] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

To help us debug your issue please explain:

- What you were trying to do (and why)

Install unar, for decompressing .rar/.7z/.zip files.

- What happened (include command output)

something happened when it's executing "xcodebuild" command... it's not a valid command on my system.

```
control@server:~$ xcodebuild
xcodebuild: command not found
```

whole terminal output (with --verbose flag): https://gist.github.com/axzxc1236/b964482a22693836298dfb3bd9b920ea

- What you expected to happen

Ugh... install successfully.

- Step-by-step reproduction instructions (by running `brew install` commands)

I ran "brew install unrar unar", terminal output: https://gist.github.com/axzxc1236/fed89c8a6fb1ad7ceb1387965231fe20

**Comment by jonchang** [comment]
This formula requires xcode to build, which is not available on Linux. I have a general fix in mind which should improve the messaging on this but I won't be able to implement it until this weekend. In the meantime please consider using `unrar` instead, which does work on linux and should be able to extract rar and 7z files.

**Comment by axzxc1236** [comment]
I tried to use unrar to extract .7z before, it only accepts .rar files.

**Comment by issyl0** [comment]
I found that the `unar` CLI [does support building on Linux](https://theunarchiver.com/command-line) it just requires some changes to the formula. Notably some more `depends_on` lines and another `make` command. Would someone like to give this a try? Otherwise I may get to it this weekend. :pray:

**Comment by issyl0** [comment]
I [made a branch for fixing the `unar` build on Linux](https://github.com/issyl0/linuxbrew-core/tree/unar-fix-build). It's missing `Foundation.h`, which is included in the `gnustep-base` package which Homebrew doesn't have as these libraries ship with macOS (I assume). I'll try to make a formula for that and then hopefully we might get further towards fixing this!

**Comment by fxcoudert** [comment]
I think it would be totally reasonable to remove `unar` from Homebrew internals (we don’t have `sit` archives, which is the only thing it was used for… what a trip down memory lane)

**Comment by issyl0** [comment]
I went [quite far down the path of fixing this](https://github.com/Homebrew/linuxbrew-core/compare/master...issyl0:unar-fix-build), yet the build for both the required formulae is still failing, I have stopped, given [previous struggles to build this on macOS](https://github.com/Homebrew/homebrew-core/pull/33176) and the fact it's not had a new version in some time.

The original reporter has installed `unar` via `apt`, which is a totally reasonable course of action. :smile:

The Homebrew code that uses `unar` clearly hasn't affected the Linux side of things and is irrelevant on macOS now. We could remove it, @fxcoudert.

I'll leave my `unar-fix-build` branch (linked above) around for historical and contextual purposes and if anyone knows exactly how to fix it in the future and wants to (:thought_balloon:), but apart from that I'm going to work on other things and close this issue.
